app/main.py

```diff
@@ -1,3 +1,21 @@
-def cache(func):
-    # Write your code here
-    pass
+import functools
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> str:
+    store = {}
+
+    @functools.wraps(func)
+    def inner(*args: Any) -> Any:
+        if f"{func.__name__}" not in store.keys():
+            store.update({f"{func.__name__}": {args: func(*args)}})
+            print("Calculating new result")
+        else:
+            if args not in store[f"{func.__name__}"]:
+                store[f"{func.__name__}"].update({args: func(*args)})
+                print("Calculating new result")
+            else:
+                print("Getting from cache")
+        return store[f"{func.__name__}"].get(args)
```

Comment:
this can be done in more simple way
you don`t need to store the function

Reply:
Please, check changes